src/include/duckdb/parser/column_definition.hpp

```diff
@@ -39,4 +39,15 @@ class ColumnDefinition {
 	DUCKDB_API static ColumnDefinition Deserialize(Deserializer &source);
 };
 
+//! A logical type definition in conjunction with any constraints implied by the type
+class ConstrainedLogicalType {
```

Comment:
I created this b/c I needed some richer info passed back up the stack from the `TransformTypeName` function- I think it could also be a mechanism to implement the check constraints on varchar(n) types and I have some steps in that direction in this PR, but I'll obviously I'd be inclined to save the real work to implement those checks for a separate PR.

Reply:
Check constraints for `VARCHAR(n)` columns can be quite easily created by just adding a `CHECK(LENGTH(col)<n)` constraint, no? Perhaps the `SERIAL` type can also be communicated back up the tree with a new constraint type `SerialConstraint` (or similar)?